elements/pl-dataframe/pl-dataframe.py

```diff
@@ -0,0 +1,164 @@
+    # TODO should there be a different display varname?
+    varname = pl.get_string_attrib(element, "params-name")
+    show_index = pl.get_boolean_attrib(element, "show-index", SHOW_INDEX_DEFAULT)
+    show_header = pl.get_boolean_attrib(element, "show-header", SHOW_HEADER_DEFAULT)
+    show_python = pl.get_boolean_attrib(element, "show-python", SHOW_PYTHON_DEFAULT)
+    show_dimensions = pl.get_boolean_attrib(
+        element, "show-dimensions", SHOW_DIMENSIONS_DEFAULT
+    )
+    show_dtype = pl.get_boolean_attrib(element, "show-dtype", SHOW_DTYPE_DEFAULT)
+
+    display_language = pl.get_enum_attrib(
+        element, "display-language", DisplayLanguage, DISPLAY_LANGUAGE_DEFAULT
+    )
+
+    num_digits = pl.get_integer_attrib(element, "digits", NUM_DIGITS_DEFAULT)
+
+    if varname not in data["params"]:
+        raise KeyError(
+            f'Could not find {varname} in params. Please make sure to set params-name="{varname}" in the element.'
+        )
+
+    # Always assume that entry in params dict is serialized dataframe
+    frame = pl.from_json(data["params"][varname])
+
+    if not isinstance(frame, pd.DataFrame):
+        raise ValueError(f"Parameter name '{varname}' does not encode a dataframe.")
+
+    frame = cast(pd.DataFrame, frame)
+
+    frame_style = frame.style
+
+    # Format integers using commas every 3 digits
+    integer_column_names = frame.select_dtypes(include="int").columns
+    frame_style.format(subset=integer_column_names, thousands=",")
+
+    if num_digits is not None:
+        # Get headers for all floating point columns and style them to use the desired number of sig figs.
+        float_column_names = frame.select_dtypes(include="float").columns
+
+        # This format string displays the desired number of digits, as given by the instructor
+        # Switches between exponential and decimal notation as needed
+        frame_style.format(subset=float_column_names, formatter=f"{{:.{num_digits}g}}")
+
+    if show_dtype:
+        if display_language is DisplayLanguage.PYTHON:
+            get_dtype_function = get_pandas_dtype
+        elif display_language is DisplayLanguage.R:
+            get_dtype_function = convert_pandas_dtype_to_r
+        else:
+            assert_never(display_language)
+
+        # old agg function:
+        descriptors = frame.agg([get_dtype_function]).set_axis(
+            ["dtype"], axis="index", copy=False
+        )
+        other = descriptors.style.applymap(lambda v: "font-weight: bold;")
+        frame_style.set_table_styles(
+            [{"selector": ".foot_row0", "props": "border-top: 1px solid black;"}]
+        )
+        frame_style.concat(other)
+
+    if not show_header:
+        frame_style.hide(axis="columns")
+
+    if show_index:
+        # Switch row indices to being 1-based if supplied indices are integer. Otherwise, leave untouched.
+        # TODO double check if this will cause unexpected behavior if these aren't just indices.
```

Comment:
👀 do you intend to resolve this TODO in this PR?

Reply:
Yes, turns out I messed up the config file a little bit and the typechecker wasn't running over this file before... Will resolve shortly.

Reply:
This has now been resolved!